lib/neo4j/active_node/query/query_proxy_eager_loading.rb

```diff
@@ -2,58 +2,153 @@ module Neo4j
   module ActiveNode
     module Query
       module QueryProxyEagerLoading
-        def each(node = true, rel = nil, &block)
-          return super if with_associations_spec.size.zero?
+        class IdentityMap < Hash
+          def add(node)
+            self[node.neo_id] ||= node
+          end
+        end
+
+        class AssociationTree < Hash
+          attr_accessor :model, :name, :association, :path
+
+          def initialize(model, name = nil)
+            super()
+            self.model = name ? target_class(model, name) : model
+            self.name = name
+            self.association = name ? model.associations[name] : nil
+          end
 
-          query_from_association_spec.pluck(identity, "[#{with_associations_return_clause}]").map do |record, eager_data|
-            eager_data.each_with_index do |eager_records, index|
-              record.association_proxy(with_associations_spec[index]).cache_result(eager_records)
+          def clone
+            super.tap { |copy| copy.each { |key, value| copy[key] = value.clone } }
+          end
+
+          def add_association(spec)
+            if spec.is_a?(Array)
+              spec.each { |s| add_association(s) }
+            elsif spec.is_a?(Hash)
+              spec.each { |k, v| (self[k] ||= AssociationTree.new(model, k)).add_association(v) }
+            else
+              self[spec] ||= AssociationTree.new(model, spec)
             end
+          end
```

Comment:
Probably this method could be called `add_spec`.  The singular "association" seems weird when you're often adding more than one via array or hash